tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15360,8 +15360,30 @@ void DiagnoseNodeEntry(Sema &S, FunctionDecl *FD, HLSLShaderAttr *Attr) {
 
 void DiagnoseEntry(Sema &S, FunctionDecl *FD) {
   auto Attr = FD->getAttr<HLSLShaderAttr>();
-  if (!Attr)
-    return;
+  if (!Attr) {
+    // if this is the Entry FD, then add the target profile
```

Comment:
Can we wrap this code in a function, then unconditionally add the attribute before this function looks it up?  Otherwise, this code will ignore if we specify a target that conflicts with an existing shader attribute.